samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -37,98 +44,166 @@
 import org.apache.samza.execution.ExecutionPlan;
 import org.apache.samza.execution.ExecutionPlanner;
 import org.apache.samza.execution.StreamManager;
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.operators.BaseTableDescriptor;
 import org.apache.samza.operators.OperatorSpecGraph;
-import org.apache.samza.operators.StreamGraphSpec;
+import org.apache.samza.table.TableConfigGenerator;
+import org.apache.samza.table.TableSpec;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 /**
  * Defines common, core behavior for implementations of the {@link ApplicationRunner} API.
  */
-public abstract class AbstractApplicationRunner extends ApplicationRunner {
+public abstract class AbstractApplicationRunner implements ApplicationRunner {
   private static final Logger log = LoggerFactory.getLogger(AbstractApplicationRunner.class);
 
+  protected final AppDescriptorImpl appDesc;
+  protected final Config config;
+  protected final Map<String, MetricsReporter> metricsReporters = new HashMap<>();
+
   /**
-   * The {@link ApplicationRunner} is supposed to run a single {@link StreamApplication} instance in the full life-cycle
+   * This is a temporary helper class to include all common logic to generate {@link JobConfig}s for high- and low-level
+   * applications in {@link LocalApplicationRunner} and {@link RemoteApplicationRunner}.
+   *
+   * TODO: Fix SAMZA-1811 to consolidate the planning into {@link ExecutionPlanner}
    */
-  protected final StreamGraphSpec graphSpec;
+  static abstract class JobPlanner {
 
-  public AbstractApplicationRunner(Config config) {
-    super(config);
-    this.graphSpec = new StreamGraphSpec(config);
-  }
+    protected final AppDescriptorImpl appDesc;
+    protected final Config config;
```

Comment:
In the newly updated ApplicationRunners, config is not accessed at all. The config is now only accessed in JobPlanner classes. I would leave it there since in the ApplicationRunner context, there is no confusion now. Configs are generated by JobPlanner.prepareJobs() now.